design.md

```diff
@@ -0,0 +1,90 @@
+# Design Considerations:
+**Task:** port azuremetadata.pl (perl) to python
+
+**Requirements:**
+(Or rather, the Methodology used to Generate requirements):
+1. compare azuremetadata CLI with the other metadata CLI modules for other cloud providers (ec2 and gce)
+2. Identify and interview current users of metadata modules (Sean)
+
+**Observations:**
+- The 3 CLI metadata modules appear to have been written at 3 different times by 3 different
+authors.
+- Subtle and not so subtle differences exist in the CLI behaviors and output formats.
+This requires developers to implement separate handlers and be aware of the idiosyncrasies
+of each implementation.
+- The GCE and EC2 modules both use getopt() and are largely familiar API except for
+the returned results.
+- Azure (the target of this port) is expectedly weird with different API, results, and configuration parameters
+required (...and because it was written in Perl)
+- It looks like the list of available and supported CLI query options for each cloud provider
+has been added to over time.
+To allow a new item to be queried from the CLI, the cloud module must also
+be revised in order to access it.
+- There is also a practice of adding other SUSE specific functions into the "cloud metadata"
+rather than breaking them into separate CLIs. e.g "cloud-service", "billing-tag", etc.
```

Comment:
The "billing-tag" is not a SUSE specific feature it is used by other image providers in Azure. There are no SUSE unique/specific features in any of the existing metadata tool implementations.